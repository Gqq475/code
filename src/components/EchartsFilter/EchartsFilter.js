import React from 'react'
import classes from './EchartsFilter.scss'
import DatePicker2 from 'components/DatePicker2'
import AvailableFilters from 'components/AvailableFilters'
import { Button } from '@blueprintjs/core'
import { dateHOC } from '../../Utils'
import { assign } from 'lodash'

const formatDate = (date) => date.format('YYYY-MM-DD HH:mm:ss')

type Props = {
  dateStart: String,
  dateEnd: String,
  data: Object,
  getEchartsLeftMenu: Function,
  getHandlingPath: Function,
  getEquipmentPort: Function,
  getSTK: Function,
  getFromTo: Function
};
export class EchartsFilter extends React.Component {
  props: Props;

  componentWillMount () {
    this.state = {
      // start: new Date(2017, 0, 1),
      // end: new Date(),
      product_code: 'SV509-CF',
      operation_code: '2100',
      line_code: 'PIPR01',
      sub_eq_code: 'PIN02'
    }
    this.props.getEchartsLeftMenu()
    this.handClick = this.handClick.bind(this)
    this.handClick()
  }

  // componentWillReceiveProps (nextProps) {
  //   if (this.props.data && this.props.data.echartsLeftMenu && this.props.data.echartsLeftMenu.result) {
  //     this.setState({
  //       product_code: this.state.product_code || this.props.data.echartsLeftMenu.result.product_code[0],
  //       operation_code: this.state.operation_code || this.props.data.echartsLeftMenu.result.operation_code[0],
  //       line_code: this.state.line_code || this.props.data.echartsLeftMenu.result.line_code[0],
  //       sub_eq_code: this.state.sub_eq_code || this.props.data.echartsLeftMenu.result.sub_eq_code[0]
  //     })
  //   }
  // }

  handClick () {
    const tmp = assign({}, {
      start_time: formatDate(this.props.dateStart),
      end_time: formatDate(this.props.dateEnd)
    }, this.state)

    this.props.getHandlingPath(tmp)
    this.props.getEquipmentPort(tmp)
    this.props.getSTK(tmp)
    this.props.getFromTo(tmp)
  }

  substrString (date) {
    let dateTime = date
    if (date < 10 && date > 0) {
      dateTime = date.substr(1, 1)
    }
    return dateTime
  }

  render () {
    return (
      <div className={classes['EchartsLeftMenu-container']}>
        <DatePicker2 {...this.props} />
        <AvailableFilters
          {...this.state}
          that={this}
          productArr={this.props.data &&
            this.props.data.echartsLeftMenu && this.props.data.echartsLeftMenu.result.product_code}
          operationArr={this.props.data &&
            this.props.data.echartsLeftMenu && this.props.data.echartsLeftMenu.result.operation_code}
          lineIDArr={this.props.data &&
            this.props.data.echartsLeftMenu && this.props.data.echartsLeftMenu.result.line_code}
          subEntityArr={this.props.data &&
            this.props.data.echartsLeftMenu && this.props.data.echartsLeftMenu.result.sub_eq_code}
          />
        <Button className={classes['btn-submit']} onClick={this.handClick}>
          查询
        </Button>
      </div>
    )
  }
}

export default dateHOC(props => <EchartsFilter {...props} />)
