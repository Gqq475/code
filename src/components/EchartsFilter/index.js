import EchartsFilter from './EchartsFilter'
export default EchartsFilter
