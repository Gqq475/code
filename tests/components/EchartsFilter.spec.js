import React from 'react'
import moment from 'moment'
import { EchartsFilter } from 'components/EchartsFilter/EchartsFilter'

describe('(Component) EchartsFilter', () => {
  it('render normally', () => {
    const getEchartsLeftMenu = jest.fn()
    const getHandlingPath = jest.fn()
    const getEquipmentPort = jest.fn()
    const getSTK = jest.fn()
    const getFromTo = jest.fn()
    const wrapper = shallow(
      <EchartsFilter
        dateStart={moment('2017-06-07T17:12:35+08:00')}
        dateEnd={moment('2017-06-07T17:12:35+08:00')}
        getEchartsLeftMenu={getEchartsLeftMenu}
        getHandlingPath={getHandlingPath}
        getEquipmentPort={getEquipmentPort}
        getSTK={getSTK}
        getFromTo={getFromTo} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
