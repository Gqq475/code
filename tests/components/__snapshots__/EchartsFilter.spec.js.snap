// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) EchartsFilter render normally 1`] = `
<div
  className="EchartsLeftMenu-container"
>
  <DatePicker2
    dateEnd={"2017-06-07T09:12:35.000Z"}
    dateStart={"2017-06-07T09:12:35.000Z"}
    getEchartsLeftMenu={[Function]}
    getEquipmentPort={[Function]}
    getFromTo={[Function]}
    getHandlingPath={[Function]}
    getSTK={[Function]}
  />
  <AvailableFilters
    line_code="PIPR01"
    operation_code="2100"
    product_code="SV509-CF"
    sub_eq_code="PIN02"
    that={
      EchartsFilter {
        "_reactInternalInstance": ShallowComponentWrapper {
          "_calledComponentWillUnmount": false,
          "_compositeType": 0,
          "_context": Object {},
          "_currentElement": <EchartsFilter
            dateEnd={"2017-06-07T09:12:35.000Z"}
            dateStart={"2017-06-07T09:12:35.000Z"}
            getEchartsLeftMenu={[Function]}
            getEquipmentPort={[Function]}
            getFromTo={[Function]}
            getHandlingPath={[Function]}
            getSTK={[Function]}
      />,
          "_debugID": 1,
          "_hostContainerInfo": null,
          "_hostParent": null,
          "_instance": [Circular],
          "_mountOrder": 1,
          "_pendingCallbacks": null,
          "_pendingElement": null,
          "_pendingForceUpdate": false,
          "_pendingReplaceState": false,
          "_pendingStateQueue": null,
          "_renderedComponent": NoopInternalComponent {
            "_currentElement": <div
              className="EchartsLeftMenu-container"
      >
              <DatePicker2
                      dateEnd={"2017-06-07T09:12:35.000Z"}
                      dateStart={"2017-06-07T09:12:35.000Z"}
                      getEchartsLeftMenu={[Function]}
                      getEquipmentPort={[Function]}
                      getFromTo={[Function]}
                      getHandlingPath={[Function]}
                      getSTK={[Function]}
              />
              <AvailableFilters
                      lineIDArr={undefined}
                      line_code="PIPR01"
                      operationArr={undefined}
                      operation_code="2100"
                      productArr={undefined}
                      product_code="SV509-CF"
                      subEntityArr={undefined}
                      sub_eq_code="PIN02"
                      that={[Circular]}
              />
              <Blueprint.Button
                      className="btn-submit"
                      onClick={[Function]}
              >
                      查询
              </Blueprint.Button>
      </div>,
            "_debugID": 2,
            "_renderedOutput": <div
              className="EchartsLeftMenu-container"
      >
              <DatePicker2
                      dateEnd={"2017-06-07T09:12:35.000Z"}
                      dateStart={"2017-06-07T09:12:35.000Z"}
                      getEchartsLeftMenu={[Function]}
                      getEquipmentPort={[Function]}
                      getFromTo={[Function]}
                      getHandlingPath={[Function]}
                      getSTK={[Function]}
              />
              <AvailableFilters
                      lineIDArr={undefined}
                      line_code="PIPR01"
                      operationArr={undefined}
                      operation_code="2100"
                      productArr={undefined}
                      product_code="SV509-CF"
                      subEntityArr={undefined}
                      sub_eq_code="PIN02"
                      that={[Circular]}
              />
              <Blueprint.Button
                      className="btn-submit"
                      onClick={[Function]}
              >
                      查询
              </Blueprint.Button>
      </div>,
          },
          "_renderedNodeType": 0,
          "_rootNodeID": 0,
          "_topLevelWrapper": null,
          "_updateBatchNumber": null,
          "_warnedAboutRefsInRender": false,
        },
        "context": Object {},
        "handClick": [Function],
        "props": Object {
          "dateEnd": "2017-06-07T09:12:35.000Z",
          "dateStart": "2017-06-07T09:12:35.000Z",
          "getEchartsLeftMenu": [Function],
          "getEquipmentPort": [Function],
          "getFromTo": [Function],
          "getHandlingPath": [Function],
          "getSTK": [Function],
        },
        "refs": Object {},
        "state": Object {
          "line_code": "PIPR01",
          "operation_code": "2100",
          "product_code": "SV509-CF",
          "sub_eq_code": "PIN02",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      }
    }
  />
  <Blueprint.Button
    className="btn-submit"
    onClick={[Function]}
  >
    查询
  </Blueprint.Button>
</div>
`;
